ably/realtime/realtime.py

```diff
@@ -83,7 +88,20 @@ def release(self, name):
         del self.all[name]
 
     def on_channel_message(self, msg):
+        channel_name = msg.get('channel')
+        if not channel_name:
+            log.error(
+                'Channels.on_channel_message()',
+                f'received event without channel, action = {msg.get("action")}'
+            )
+            return
+
         channel = self.all.get(msg.get('channel'))
         if not channel:
-            log.warning('Channel message recieved but no channel instance found')
+            log.warning(
+                'Channels.on_channel_message()',
+                f'receieved event for non-existent channel: {channel_name}'
+            )
+            return
```

Comment:
This should probably be an error since it terminates here. A warning will indicate it continues despite the condition

Reply:
This method is not public API - this condition is just for if the library receives a protocol message for a channel which is not known to the client, so the client will indeed continue operating if it happens